tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp

```diff
@@ -360,3 +387,269 @@ TEST_F(CRTLoggingTest, testTraceLogLevel)
 {
     DoCRTLogTest(LogLevel::Trace);
 }
+
+class LoggingTestLogFileRace : public Aws::Testing::AwsCppSdkGTestSuite
+{
+};
+
+void LogOnCRTLogSystemInterfaceWithoutVaArgs(CRTLogSystemInterface* object, LogLevel logLevel, const char* subjectName, const char* formatStr, ...)
+{
+    va_list args;
+    va_start(args, formatStr);
+    assert(object);
+    object->Log(logLevel, subjectName, formatStr, args);
+    va_end(args);
+}
+
+TEST_F(LoggingTestLogFileRace, testRaceOnLogFile)
+{
+    /*
+     * Tests a race condition on the default log file (in case of multiple processes / loggers using the same SDK log file).
+     */
+    static const size_t PARALLEL_COUNT = 32;
+    Aws::Vector<std::future<bool>> futures(PARALLEL_COUNT);
+
+    std::atomic<bool> run;
+    run = false;
+
+    for(size_t i = 0; i < PARALLEL_COUNT; ++i)
+    {
+        futures[i] = std::async(std::launch::async,
+                [i, &run]()-> bool
+                {
+                    while(true)
+                    {
+                        if(run.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    Aws::UniquePtr<DefaultLogSystem> logSystem =
+                            Aws::MakeUnique<DefaultLogSystem>(AllocationTag, LogLevel::Info, "testRaceOnLogFile");
+                    Aws::String logMsgTag = "testRaceOnLogFile thread #" + Aws::Utils::StringUtils::to_string(i);
+                    logSystem->Log(LogLevel::Info, logMsgTag.c_str(), "Knock knock");
+                    logSystem->Log(LogLevel::Info, logMsgTag.c_str(), " - Who's there?");
+                    logSystem->Log(LogLevel::Info, logMsgTag.c_str(), " - Race condition!");
+                    for(size_t j = 0; j < 100; ++j)
+                    {
+                        logSystem->Log(LogLevel::Info, logMsgTag.c_str(), "All work and no play makes Jack a dull boy");
+                    }
+
+                    return true;
+                });
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
dont we want to sleep after setting run to true? we are running those threads in busy while loops during the wait otherwise?

Reply:
future.get() will block main thread.
This sleep is just to wait a bit after submitting a bunch of threads to let them prepare and fall into loop.